### ISSUE TITLE
"open this in GitHub Desktop" link in pull requests is broken

### ISSUE THREAD

**Issue by kylefox** [desktop/desktop#5913]
## Description

Clicking the "open this in GitHub Desktop" link in pull requests refuses to open the pull request — instead, it constantly re-prompts me to clone the repository.

![image](https://user-images.githubusercontent.com/37838/47040004-770a2380-d142-11e8-999a-7b18fe0c169b.png)

![image](https://user-images.githubusercontent.com/37838/47040160-d49e7000-d142-11e8-9a9e-d476dcc1ac8d.png)

I have tried deleting and re-cloning numerous times, but the issue still persists.

## Version

* GitHub Desktop: 1.4.2

* Operating system: macOS Mojave (10.14)

## Steps to Reproduce

1. Locate a pull request on GitHub _for a repo that you've already cloned into GitHub Desktop._
2. Click the "open this in GitHub Desktop" link at the bottom of the pull request.

### Expected Behavior

GitHub Desktop opens the pull request (i.e. that branch is checked out).

### Actual Behavior

The GitHub Desktop app gains focus and prompts you to clone the repository:

![image](https://user-images.githubusercontent.com/37838/47040160-d49e7000-d142-11e8-9a9e-d476dcc1ac8d.png)

### Logs

[2018-10-16.desktop.production.log](https://github.com/desktop/desktop/files/2484548/2018-10-16.desktop.production.log)

**Comment by shiftkey** [comment]
@kylefox thanks for the report!

I've been able to reproduce this and I believe it's down to this chunk of code and a change we made in #5576:

https://github.com/desktop/desktop/blob/40b3d53b3f9a3d3ddf9ac6efddfe8db58bf67511/app/src/lib/dispatcher/dispatcher.ts#L976-L986

The `gitHubRepository.cloneURL === url` check here now fails to find a match because we no longer prefix a `.git` to the URLs we get from GitHub. And because never match your existing repository, we ask you to clone it again.
